src/components/shared/AutoSubmit.tsx

```diff
@@ -0,0 +1,41 @@
+import { debounce } from '@mui/material'
+import { useFormikContext } from 'formik'
+import React, { useCallback } from 'react'
+
+const SubmitListener = () => {
+  const formik = useFormikContext()
+  const [lastValues, updateState] = React.useState(formik.values)
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const submitForm = useCallback(
+    debounce((): void => {
+      formik.submitForm()
+    }, 500),
+    []
+  )
+
+  React.useEffect(() => {
```

Comment:
We should think of a better way for doing this.

Reply:
What do you suggest ?